src/netaddress.cpp

```diff
@@ -36,10 +37,20 @@ void CNetAddr::SetRaw(Network network, const uint8_t *ip_in)
     switch(network)
     {
         case NET_IPV4:
+            m_net = NET_IPV4;
             memcpy(ip, pchIPv4, 12);
             memcpy(ip+12, ip_in, 4);
             break;
         case NET_IPV6:
+            if (memcmp(ip_in, pchIPv4, sizeof(pchIPv4)) == 0) {
```

Comment:
I think it would make sense to add this detection out to a separate function, e.g. `SetLegacyIPV6` then call it here and in the deserialization logic, highlighting that it does auto-detection.

Reply:
Done: https://github.com/bitcoin/bitcoin/compare/662bb25efa0d164edace0cd2809ba85e5baf19fe..64897c5776f024662000528415315995e2a83d3c